servicetalk-opentelemetry-http/src/test/java/io/servicetalk/opentelemetry/http/OpenTelemetryHttpRequestFilterTest.java

```diff
@@ -248,6 +257,110 @@ void testCaptureHeader() throws Exception {
         }
     }
 
+    @Test
+    void transportObserver() throws Exception {
```

Comment:
Consider also enhancing `AsyncContextCaptureConnectionObserver` inside `HttpTransportObserverAsyncContextTest` to assert `AsyncContext` where it's available